babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -788,15 +788,71 @@ SqlIndex SqlCreateTableIndex(Span s) :
 SqlNode SqlExecMacro() :
 {
     SqlIdentifier macro;
+    SqlNodeList paramNames = new SqlNodeList(getPos());
+    SqlNodeList paramValues = new SqlNodeList(getPos());
     Span s;
 }
 {
     macro = CompoundIdentifier() { s = span(); }
+    [
+        SqlExecMacroArgument(paramNames, paramValues)
+    ]
     {
-        return new SqlExecMacro(s.end(this), macro);
+        return new SqlExecMacro(s.end(this), macro, paramNames, paramValues);
     }
 }
 
+void SqlExecMacroArgument(SqlNodeList paramNames, SqlNodeList paramValues) :
+{
+    SqlNode name;
+    SqlNode value;
+}
+{
+        <LPAREN>
+        (
+            value = Literal()
+            {
+                paramValues.add(value);
+            }
+
+        )
+        (
+            <COMMA>
+            value = Expression(ExprContext.ACCEPT_SUB_QUERY)
+            {
+                paramValues.add(value);
+            }
+
+        )*
+        <RPAREN>
+    |
+        <LPAREN>
+        (
+            name = SimpleIdentifier()
+            <EQ>
+            {
+                paramNames.add(name);
+            }
+            value = Expression(ExprContext.ACCEPT_SUB_QUERY)
+            {
+                paramValues.add(value);
+            }
+        )
+        (
+            <COMMA>
+            name = SimpleIdentifier()
+            <EQ>
+            {
+                paramNames.add(name);
+            }
+            value = Expression(ExprContext.ACCEPT_SUB_QUERY)
+            {
+                paramValues.add(value);
+            }
+        )*
+        <RPAREN>
```

Comment:
Since you have repeated logic, it may be worth considering extracting it to a separate function, passing the paramNames and paramValues list to that function, and have it handle adding things to the lists. Then you could have 
something like this (not necessarily with same func name):
extractParamPair(paramNames, paramValues)
( 
  <COMMA> 
  extractParamPair(paramNames, paramValues)
)*
This could also be useful later down the line if it turns out this kind of "paramName = paramValue" syntax appears in other contexts.

Reply:
This is a good point. Now the code does not work on two separated list anymore, I think it would be reasonable to leave it as it is. There are still few token that are repeated. However since only three token are repeated, contributors can extract the logic when a future feature needs it.